tests/python_tests/test_vlm_pipeline.py

```diff
@@ -1167,17 +1175,31 @@ def get_nanollava_processor():
             ],
         }
     ]
-    if video_name is not None:
-        prompt = "Describe this video."
-        resized_video = request.getfixturevalue(video_name)
-        conversation[0]["content"] = [{"type": "video"}] + conversation[0]["content"]
-    if image_name is not None:
-        resized_image = request.getfixturevalue(image_name)
-        if video_name is not None:
-            prompt = "Describe this image and video."
-        else:
+
+    if model_id in ["katuni4ka/tiny-random-qwen2.5-vl", "katuni4ka/tiny-random-qwen2vl"]:
+        if image_name is not None:
+            resized_image = request.getfixturevalue(image_name)
             prompt = "Describe this image."
-        conversation[0]["content"] = [{"type": "image"}] + conversation[0]["content"]
+            conversation[0]["content"] = [{"type": "image"}] + conversation[0]["content"]
+        if video_name is not None:
+            if image_name is not None:
+                prompt = "Describe this image and video."
+            else:
+                prompt = "Describe this video."
+            resized_video = request.getfixturevalue(video_name)
+            conversation[0]["content"] = [{"type": "video"}] + conversation[0]["content"]
+    else:
+        if video_name is not None:
+            prompt = "Describe this video."
+            resized_video = request.getfixturevalue(video_name)
+            conversation[0]["content"] = [{"type": "video"}] + conversation[0]["content"]
+        if image_name is not None:
+            resized_image = request.getfixturevalue(image_name)
+            if video_name is not None:
+                prompt = "Describe this image and video."
+            else:
+                prompt = "Describe this image."
+            conversation[0]["content"] = [{"type": "image"}] + conversation[0]["content"]
```

Comment:
This if-statement logic seems to be a duplication

Reply:
You are right, remove duplicated part, still need to reverse order of image and video in the qwen-vl series model.
updated.